test/gtest/conv3d_test.cpp

```diff
@@ -0,0 +1,158 @@
+    {
+    case miopenFloat: params = ConfigWithFloat::GetParam(); break;
+    case miopenInt8:
+    case miopenHalf:
+    case miopenBFloat16:
+    case miopenInt32:
+    case miopenDouble:
+        FAIL() << "miopenHalf, miopenBFloat16, miopenInt8, miopenInt32, "
+                  "miopenDouble data "
+                  "type not supported by "
+                  "test_conv_hip_igemm_xdlops test";
+
+    default: params = ConfigWithFloat::GetParam();
+    }
+
+    for(const auto& test_value : params)
+    {
+        std::vector<std::string> tokens;
+        GetArgs(test_value, tokens);
+        std::vector<const char*> ptrs;
+
+        std::transform(tokens.begin(), tokens.end(), std::back_inserter(ptrs), [](const auto& str) {
+            return str.data();
+        });
+
+        testing::internal::CaptureStderr();
+        test_drive<conv3d_driver>(ptrs.size(), ptrs.data());
+        auto capture = testing::internal::GetCapturedStderr();
+        std::cout << capture;
+    }
+};
+
+bool IsTestSupportedForDevice(const miopen::Handle& handle)
+{
+    std::string devName = handle.GetDeviceName();
+    if(devName == "gfx94" || devName == "gfx103" || devName == "gfx110")
+        return true;
+    else
+        return false;
+}
+
+std::vector<std::string> GetTestCases(const std::string& precision)
+{
+    std::string psd0 = " --pads_strides_dilations 0 0 0 1 1 1 1 1 1";
+    std::string psd1 = " --pads_strides_dilations 0 0 0 2 2 2 1 1 1";
+    std::string psd2 = " --pads_strides_dilations 2 2 2 1 1 1 1 1 1";
+    std::string psd3 = " --pads_strides_dilations 0 0 0 1 1 1 2 2 2";
+    std::string psd4 = " --pads_strides_dilations 1 1 1 1 1 1 1 1 1";
+    std::string psd5 = " --pads_strides_dilations 0 0 0 1 1 1 1 1 1";
+
+    std::vector<std::string> test_cases = {
+        // clang-format off
+    // test_conv3d_extra
+    {precision + "--input 2 16 50 50 50 --weights 32 16 5 5 5" + psd0},
+    {precision + "--input 2 16 50 50 50 --weights 32 16 5 5 5" + psd1},
+    {precision + "--input 2 16 50 50 50 --weights 32 16 5 5 5" + psd2},
+    {precision + "--input 2 16 50 50 50 --weights 32 16 5 5 5" + psd3},
+    //test_conv3d_extra reduced set
+    {precision + "--input 2 16 50 50 50 --weights 32 16 5 5 5" + psd0 },
+    {precision + "--input 2 16 50 50 50 --weights 32 16 5 5 5" + psd1 },
+    {precision + "--input 2 16 50 50 50 --weights 32 16 5 5 5" + psd2 },
+    {precision + "--input 2 16 50 50 50 --weights 32 16 5 5 5" + psd3 },
+    {precision + "--input 1 16 4 161 700 --weights 16 16 3 11 11" + psd4 },
+    {precision + "--input 1 16 4 161 700 --weights 16 16 3 11 11" + psd5 },
+    {precision + "--input 1 16 4 140 602 --weights 16 16 3 11 11" + psd4 },
+    {precision + "--input 1 16 4 140 602 --weights 16 16 3 11 11" + psd5 }
+        // clang-format on
+    };
+    return test_cases;
+}
+
+} // namespace conv3d_test
+
+using namespace conv3d_test;
+
+TEST_P(ConfigWithFloat, FloatTest_conv3d_test)
+{
+#if MIOPEN_BACKEND_OPENCL
```

Comment:
[Q] Why we need to check MIOPEN_BACKEND_OPENCL? We do not maintain this backend anymore.

Reply:
[resolved]